requirements/requirements.in

```diff
@@ -0,0 +1,110 @@
+attrs==18.1.0
+# needed to build gevent from source
+cython==0.27.3
+restkit==4.2.2
+iso8601==0.1.11
+jsonobject==0.9.1
+jsonobject-couchdbkit==0.9.1
+celery==3.1.18
+# required by django-digest
+decorator==4.0.11
+django==1.11.14
+django-braces==1.8.1
+django-celery==3.2.1
+django-crispy-forms==1.7.0
+jsonfield==1.0.3
+dropbox==7.3.1
+defusedxml==0.4.1
+diff-match-patch==20120106
+dimagi-memoized==1.0.1
+django-tastypie==0.14.0
+djangorestframework==3.5.4
+elasticsearch==1.9.0
+eulxml==1.1.3
+ghdiff==0.4
+gunicorn==19.4
+lxml==3.4.4
+kafka-python==0.9.5
+mock==2.0.0
+openpyxl==2.5.3
+Pillow==5.2.0
+phonenumberslite==8.8.9
+psycopg2==2.7.3.1
+psycogreen==1.0
+pycrypto==2.6.1
+py-KISSmetrics==1.0.1
+pygooglechart==0.4.0
+python-dateutil==2.6.1
+python-magic==0.4.13
+pytz==2017.2
+PyYAML==3.12
+requests==2.10.0
+suds-jurko==0.6
+tropo-webapi-python==0.1.3
+Unidecode==0.04.20
+xlrd==1.0.0
+simplejson==3.11.1
+sh==1.09
+gevent==1.2.2
+greenlet==0.4.12
+numpy==1.7.1
+six==1.11.0
+socketpool==0.5.3
+markdown==2.2.1
+amqp==1.4.9
+sqlagg==0.9.0
+django-redis==4.2
+redis==2.10.5
+twilio==6.5.1
+django-countries==4.6
+reportlab==3.0
+stripe==1.67.0
+django-compressor==2.1
+django-angular==0.7.16
+Pycco==0.5.1
+django-cte==1.1.2
+jsonpath-rw==1.4.0
+pip>=6.1.1
+django-transfer==0.4
+Pygments==2.0.2
+Jinja2==2.8
+tinys3==0.1.12
+django-formtools==2.0
+django-otp==0.3.13
+django-two-factor-auth==1.6.2
+datadog==0.15.0
+django-websocket-redis==0.5.1
+django-redis-sessions==0.5.6
+SQLAlchemy==1.2.9
+alembic==0.8.6
+pyzxcvbn==0.8.0
+django-statici18n==1.3.0
+django-simple-captcha==0.5.7
+httpagentparser==1.7.8
+boto3==1.4.4
+simpleeval==0.9.5
+laboratory==0.2.0
+ConcurrentLogHandler==0.9.1
+PyGithub==1.35
+django-bulk-update==1.1.10
+csiphash==0.0.5
+json-delta==2.0
+email_validator==1.0.2
+dnspython==1.15.0
+raven==6.1.0
+quickcache==0.5.0
+zeep==1.6.0
+ethiopian-date-converter==0.1.4
+-e git+git://github.com/dimagi/architect@v0.5.7a2#egg=architect
```

Comment:
Why was ```-e``` added?  It seems to be related to [failing docker builds](https://hub.docker.com/r/dimagi/commcarehq_base/builds/bz3ndj3hfsyha8f72dhvtli/):

```
Running setup.py develop for architect

    Complete output from command /usr/local/bin/python -c "import setuptools, tokenize;__file__='/src/architect/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps --user --prefix=:

    running develop
    error: can't create or remove files in install directory
    
    The following error occurred while trying to add or remove files in the
    installation directory:
    
        [Errno 2] No such file or directory: '/vendor/lib/python2.7/site-packages/test-easy-install-1665.write-test'
    
    The installation directory you specified (via --install-dir, --prefix, or
    the distutils default setting) was:
    
        /vendor/lib/python2.7/site-packages
    
    This directory does not currently exist.  Please create it and try again, or
    choose a different installation directory (using the -d or --install-dir
    option).
```

Reply:
bump @dannyroberts

Reply:
Hey sorry for dropping the ball on replying.

I added it because `pip-compile` doesn't support non-editable installs from URLs. It failed, said as much, and told me to add `-e`.

I'm now realizing that, though I'd consider it a bug in these libraries, it may be relatively common for the `-e` install to not work or at least not work in quite the same way as a non-editable install. I recently ran into this for a pycryptodome issue as well, and after some digging and I landed on this as a fix, which I'm planning to do: https://github.com/dimagi/commcare-hq/pull/21475#issuecomment-411516542.

Reply:
Okay, this fixed it: https://github.com/dimagi/commcare-hq/pull/21490. Also closing related issue: https://github.com/dimagi/commcare-hq/issues/21319